### PR TITLE
TypeScript has implemented object rest/spread properties in v2.1.0

### PR DESCRIPTION
**Pull request by Pedro-vk**

### CODE REVIEW

File: data-esnext.js [hunk 1]
```diff
@@ -778,6 +778,7 @@ exports.tests = [
     babel: true,
     jsx: true,
     chrome58: 'flagged',
+    typescript: typescript.corejs,
```
Comment (chicoxyzzy) [inline data-esnext.js]: this should be `true`, not `typescript.corejs`

File: data-esnext.js [hunk 2]
```diff
@@ -794,6 +795,7 @@ exports.tests = [
...
```
Comment (chicoxyzzy) [inline data-esnext.js]: same here

**Review by chicoxyzzy — CHANGES_REQUESTED** [review]
please run `npm run build` and add changes to your PR

File: data-esnext.js (hunk 1 above)
Comment (ljharb) [reply data-esnext.js]: doesn't it require core.js to shim `Object.assign`?

Comment (Pedro-vk) [reply data-esnext.js]: It seems no:
https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#object-spread-and-rest

Comment (ljharb) [reply data-esnext.js]: Nothing in that link talks about it either way?

Comment (chicoxyzzy) [reply data-esnext.js]: @ljharb TS doesn't require `core.js` for that
https://www.typescriptlang.org/play/index.html#src=let%20%7B%20x%2C%20y%2C%20...z%20%7D%20%3D%20%7B%20x%3A%201%2C%20y%3A%202%2C%20a%3A%203%2C%20b%3A%204%20%7D%3B%0Alet%20n%20%3D%20%7B%20x%2C%20y%2C%20...z%20%7D%3B

**Review by chicoxyzzy — APPROVED** [review]

**Comment by chicoxyzzy** [comment]
@Pedro-vk thank you!

**Comment by Pedro-vk** [comment]
My pleasure :)

File: data-esnext.js (hunk 1 above)
Comment (ljharb) [reply data-esnext.js]: Thanks, that clarifies.

However, that transpiled code violates the spec; it doesn't check for enumerability of symbols.

**Pull request merged**